pandas/core/internals/blocks.py

```diff
@@ -1589,7 +1589,7 @@ def should_store(self, value: ArrayLike) -> bool:
 
     def set(self, locs, values):
         assert locs.tolist() == [0]
-        self.values[:] = values
+        self.values = values
```

Comment:
isnt this going to break views?

Reply:
That might be, but that would mean it was broken in 1.0 as well, right?

Reply:
Assuming the `[:]` was added after 1.0, then I guess so, yes.

Reply:
is the `[:]` (i.e. behavior in master) in a released 1.0.x?

Reply:
The PR is referenced in the issue and the top post: #32831 -> this is only in master